File: 2nd year/1st semester/MAP/ToyLanguageInterpreter/src/View/Command.java
package View;

public abstract class Command {
    private String key, description;

    Command(String key, String description) { this.key = key; this.description = description;}
    public abstract void execute() throws java.lang.InterruptedException;
    public String getKey(){return key;}
    public String getDescription(){return description;}
}

File: 2nd year/1st semester/MAP/CS_ToyLanguageInterpreter/MAP_Interpreter/View/Command.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAP_Interpreter.View
{
    public abstract class Command
    {
        private string key, description;

        public Command(string key, string description)
        {
            this.key = key;
            this.description = description;
        }

        public abstract void execute();

        public string getKey() { return this.key; }
        public string getDescription() { return this.description; }
    }
}
